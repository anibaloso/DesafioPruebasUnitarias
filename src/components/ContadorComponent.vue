<script setup>
import {ref} from 'vue'

const counter = ref(0)

function aumentar() {
    counter.value++
}
function disminuir() {
    counter.value--
}
</script>
<template>
    <div>
        <h1 >Contador : <span id="contador">{{ counter }}</span></h1>
        <button @click="aumentar" id="aumentar">Aumentar +1 </button>
        <button @click="disminuir" id="disminuir">Disminuir -1 </button>

    </div>

</template>