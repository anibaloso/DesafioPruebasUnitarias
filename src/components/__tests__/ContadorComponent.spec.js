import { describe, it, expect } from "vitest";
import ContadorComponent from "../ContadorComponent.vue";
import { mount } from "@vue/test-utils";


describe('ContadorComponent.vue', () => {
    const wrapper = mount(ContadorComponent)

    it('El Html se mantiene en las pruebas', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })


    it('Tiene un boton para aumentar el contador con texto +1', () => {
        const btnAumentar = wrapper.find('#aumentar')
        expect(btnAumentar.text()).toBe('Aumentar +1')
    })
    it('tiene un boton para reducir el contador con texto disminuir-1',()=>{
        const btnReducir =wrapper.find('#disminuir')
        expect(btnReducir.text()).toBe('Disminuir -1')
    })

    it('el contador parte en 0',()=>{
        const contador=wrapper.find('#contador')
        expect(contador.text()).toBe('0')
    })
})
