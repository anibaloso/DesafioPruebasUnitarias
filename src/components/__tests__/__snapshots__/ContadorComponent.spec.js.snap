// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContadorComponent.vue > El Html se mantiene en las pruebas 1`] = `
"<div>
  <h1>Contador : <span id="contador">0</span></h1><button id="aumentar">Aumentar +1 </button><button id="disminuir">Disminuir -1 </button>
</div>"
`;
